tests/test_inference_problem.py

```diff
@@ -0,0 +1,253 @@
+from bayes.parameters import ModelParameters as ParameterList
+from collections import OrderedDict
+import numpy as np
+
+"""
+###############################################################################
+
+                            LIBRARY CODE
+
+###############################################################################
+"""
+
+
+class Sensor:
+    def __init__(self, name, shape=(1, 1)):
+        self.name = name
+        self.shape = shape
+
+    def __repr__(self):
+        return f"{self.name} {self.shape}"
+    
+class DummySensor(Sensor):
+    def __init__(self):
+        super().__init__("Dummy")
+
+
+class NormalPrior:
+    def __init__(self, mean, sd):
+        self.mean = mean
+        self.sd = sd
+
+
+class ModelError:
+    def __call__(self, parameter_list):
+        raise NotImplementedError("Override this!")
+
+
+class LatentParameter:
+    def __init__(self):
+        self._affected_parameters = []
+        self.N = None
+        self.prior = None
+        self.start_idx = None
+
+    def _contains(self, key, name):
+        return (key, name) in self._affected_parameters
+
+    def add(self, key, name, N):
+        assert not self._contains(key, name)
+        if self.N is None:
+            self.N = N
+        else:
+            assert self.N == N
+
+        self._affected_parameters.append((key, name))
+
+    def __len__(self):
+        return len(self._affected_parameters)
+
+    def __getitem__(self, i):
+        return self._affected_parameters[i]
+
+    def __str__(self):
+        if self.N == 1:
+            idx_range = str(self.start_idx)
+        else:
+            idx_range = f"{self.start_idx}..{self.start_idx+N}"
+
+        return f"{idx_range:10} {self._affected_parameters} {self.prior}"
+
+
+class LatentParameterList:
+    def __init__(self):
+        self._all_parameter_lists = {}
+        self._latent = OrderedDict()
+        self._total_length = None
+
+    def add_parameter_list(self, parameter_list, key=None):
```

Comment:
Good point. But 'model_parameter_list' would be more suited as I came across the case where a parameter was part of the model_error -- e.g. related to scaling of the data -- but not the forward model.

Reply:
Then we should maybe change that to model error prms, or live with the fact that for specific cases that fw_model_prm is not a fully correct naming. In the Bayesian community, they often also talk about modelling, but they mean the uncertainty model, so I think it's important to avoid model as a single term.

Reply:
Well, I just had a look again. In this specific case "parameter_list" could be fine, because noise_parameter_lists or hyper_parameter_list could also be stored there.